### PR TITLE
negotiate_kerberos_auth: Fix build [-Wunused-but-set-variable]

### PR DESCRIPTION
**Pull request by kinkie**
Unused variable discovered by clang v15.

### CODE REVIEW

**Comment by kinkie** [comment]
Note: this PR will have to be force-merged because #1295 would prevent a clean merge test.

**Comment by rousskov** [comment]
> Note: this PR will have to be force-merged because #1295 would prevent a clean merge test.

Please do not force-merge any PRs, especially the currently [broken](https://github.com/squid-cache/squid/actions/runs/4309090163/jobs/7516068240) ones like this: There is no reason to enlarge the scope of the problem.

I marked the Jenkins test as optional in GitHub[^1]. If your PR is fixing a Jenkins build problem (like this one), please watch for the Jenkins test results and make sure that they do not show the specific build failures that your PR is fixing (other/old build failures are fine -- they will be presumably fixed by other PRs).

Please do not clear new PRs for merging unless they are fixing a build on Jenkins.

[^1]: We will make it required again ASAP, of course, after all PRs that fix Jenkins tests are merged.

**Comment by rousskov** [comment]
@eduard-bagdasaryan, please adjust Anubis configuration to match the [removal](https://github.com/squid-cache/squid/pull/1294#issuecomment-1451194356) of the required Jenkins tests. If a configuration adjustment is not enough to let PRs merge despite Jenkins failures, please let me know.

**Review by rousskov — APPROVED** [review]
I removed one more line that was using `count` and adjusted PR title/description (while guessing what clang warning broke the build).

**Comment by rousskov** [comment]
@kinkie, if you are OK with the current PR title/description, and the Jenkins test produces the expected outcome, please clear this PR for merging.

**Comment by kinkie** [comment]
since we have multiple reasons for build test failures, I'll have to manually land these changes, bypassing Anubis, except for the last one. Fortunately the changes are all trivial

**Comment by eduard-bagdasaryan** [comment]
> adjust Anubis configuration to match the https://github.com/squid-cache/squid/pull/1294#issuecomment-1451194356 of the required Jenkins tests

Anubis has only a setting for the number of _staging_ checks that it waits for. It requests the list of required PR checks from GitHub - so it should notice that you have made the PR test (Jenkins) optional.

**Comment by rousskov** [comment]
> since we have multiple reasons for build test failures, I'll have to manually land these changes

The plan I [suggested](https://github.com/squid-cache/squid/pull/1294#issuecomment-1451194356) should work, in principle, regardless of the number of Jenkins test failures. Please do not merge PRs manually until we are certain that there is no way to adjust GitHub and Anubis to continue to merge PRs automatically while ignoring Jenkins failures. As of now, there is no such certainty, and there is still hope that the opposite is true.

**Comment by rousskov** [comment]
> Anubis has only a setting for the number of _staging_ checks that it waits for.

I expect some of those staging tests to fail as well: Jenkins build tests (on platforms that detect bugs that have slipped into the current master code) will fail during PR branch testing _and_ during staging tests. AFAICT, we will need to either keep temporary adjusting the configuration counter to accommodate the (decreasing) number of failed staging tests or temporary adjust Anubis code to treat failed Jenkins staging tests as successful, whichever is easier.

**Comment by eduard-bagdasaryan** [comment]
> some of those staging tests to fail as well

Ok, if that list of failed staging tests is still unknown, let's wait until this PR becomes staged and then adjust that parameter accordingly.

**Comment by rousskov** [comment]
> Ok, if that list of failed staging tests is still unknown, let's wait until this PR becomes staged and then adjust that parameter accordingly.

The problem is not so much that the list (length) is currently _unknown_. The problem is that the list (length) may be _changing_ -- some PRs will fix the last bug on a specific platform. Merging those PRs will decreasing the number of failed staged tests (eventually to zero). We can either adjust configuration for every such PR or add a hack to treat Jenkins failures as successes (which is similar to what a manual merge would have to do anyway and is quite safe in these specific circumstances).

**Comment by kinkie** [comment]
> > since we have multiple reasons for build test failures, I'll have to manually land these changes
> 
> The plan I [suggested](https://github.com/squid-cache/squid/pull/1294#issuecomment-1451194356) should work, in principle, regardless of the number of Jenkins test failures. Please do not merge PRs manually until we are certain that there is no way to adjust GitHub and Anubis to continue to merge PRs automatically while ignoring Jenkins failures. As of now, there is no such certainty, and there is still hope that the opposite is true.

Yes. I wrote this before seeing your comment. Thanks for helping nudge the situation

**Review by yadij — APPROVED** [review]

**Pull request closed without merging**
